### PR TITLE
[CDEC-327] Write test utility functions for displaying hexadecimal.

### PR DESCRIPTION
**Pull request by intricate**
## Description

Write test utility functions for displaying hexadecimal.

## Linked issue

[https://iohk.myjetbrains.com/youtrack/issue/CDEC-327](https://iohk.myjetbrains.com/youtrack/issue/CDEC-327)
## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: util/test/Test/Pos/Util/Base16.hs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    -- to base-16 yet so we're not checking for <= 32), then just 'encode'
+    -- rather than prepending the byte offsets.
+    | LB.length bs <= (lineWrapLength `div` 2) = encode bs
+    | otherwise = LB.concat $ go 0 (chunkBS lineWrapLength $ B16.encode bs)
+  where
+    go :: Int64 -> [LB.ByteString] -> [LB.ByteString]
+    go _   []     = []
+    go acc (x:xs) =
+        LB.concat
+            [ (LB.pack $ printf "%08x: " acc)
```
Comment (intricate) [inline util/test/Test/Pos/Util/Base16.hs]: @parsonsmatt @erikd Opinions on the use of printf?

File: util/test/Test/Pos/Util/Base16.hs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+-- | Line wraps a ByteString every x bytes.
+lineWrapBS :: Int64 -> LB.ByteString -> LB.ByteString
+lineWrapBS n s = LB.intercalate "\n" $ chunkBS n s
+
+-- | Divides a ByteString into x-length "chunks".
+chunkBS :: Int64 -> LB.ByteString -> [LB.ByteString]
+chunkBS n xs = case LB.uncons xs of
+    Nothing -> []
+    Just _  -> LB.take n xs : chunkBS n (LB.drop n xs)
```
Comment (erikd) [inline util/test/Test/Pos/Util/Base16.hs]: Doing `LB.take` and `LB.drop` is two operations. Have a look at `BS.splitAt` :)

File: util/test/Test/Pos/Util/Base16.hs (hunk 1 above)
Comment (erikd) [reply util/test/Test/Pos/Util/Base16.hs]: The vast majority of the time, the lengths of bytestrings we're using this for will be up to maybe 1 killobyte, so 8 p;aces on the hex is probably a but much. Would be cool if the sting was less than 256 bytes it only used 2 places and used more places if required.

Comment (erikd) [reply util/test/Test/Pos/Util/Base16.hs]: I would be sqeamish about using `print` in production code, especially if it was not by a property test, but in test code I don't have much hesitation.

**Comment by intricate** [comment]
Closed in favour of #2979

**Pull request closed without merging**